src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -419,7 +419,7 @@ void SyncInferRequest::set_tensor(const ov::Output<const ov::Node>& in_port, con
         const auto& shape = port.get_partial_shape();
         const bool isDynamic = shape.is_dynamic();
 
-        if (!shape.compatible(ov::PartialShape(tensor->get_shape()))) {
+        if (!shape.compatible(ov::PartialShape(tensor->get_shape())) && tensor->get_size() != 0) {
```

Comment:
does it make sense that in general the hardware plugins can align the the common basic compatibility checking?  @ilya-lavrenov 
we have below in gpu plugin, which performs similar checking, but with different conditions in if clause:
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp#L144

Reply:
I suppose we discussed in email thread "[CPU Plugin] Infer request failed to set customer output tensor with dynamic shape" that checks should be aligned.